sidecar-injector/pkg/inject/common.go

```diff
@@ -18,74 +19,60 @@ func mutationRequired(ignoredList []string, metadata *metav1.ObjectMeta) bool {
 		}
 	}
 
-	annotations := metadata.GetAnnotations()
-	if annotations == nil {
-		annotations = map[string]string{}
-	}
-
-	status := annotations[admissionWebhookAnnotationStatusKey]
+	status, _ := getAnnotation(metadata, annotationStatusKey)
 
 	// determine whether to perform mutation based on annotation for the target resource
 	var required bool
 	if strings.ToLower(status) == "injected" {
 		required = false;
```

Comment:
You can probably eliminate the else if you initialize `required := false` and snip out the first part of the `if`

Reply:
done